src/porepy/numerics/interface_laws/contact_mechanics_interface_laws.py

```diff
@@ -0,0 +1,781 @@
+        # The mortar variable acts as a Dirichlet boundary condition for the master.
+        master_bound_stress = data_master[pp.DISCRETIZATION_MATRICES][
+            self.discr_master.keyword
+        ]["bound_stress"]
+        master_stress = data_master[pp.DISCRETIZATION_MATRICES][
+            self.discr_master.keyword
+        ]["stress"]
+        master_bc_values = data_master[pp.PARAMETERS][self.discr_master.keyword][
+            "bc_values"
+        ]
+        master_divergence = pp.fvutils.vector_divergence(g_master)
+
+        # The mortar variable (boundary displacement) takes the form of a Dirichlet
+        # condition for the master side. The MPSA convention is to have
+        # - div * bound_stress * bc_values
+        # on the rhs. Accordingly, the contribution from the mortar variable (boundary
+        # displacement) on the left hand side is positive:
+        # div * bound_stress * u_mortar
+        cc[master_ind, mortar_ind] = (
+            master_divergence
+            * master_bound_stress
+            * mg.mortar_to_master_avg(nd=ambient_dimension)
+        )
+
+        ### Equation for the slave side
+        #
+        # These are the contact conditions, which dictate relations between
+        # the contact forces on the slave, and the displacement jumps.
+        #
+        # NOTE: Both the contact conditions and the contact stresses are defined in the
+        # local coordinate system of the surface. The displacements must therefore
+        # be rotated to this local coordinate system during assembly.
+        traction_discr, displacement_jump_discr, rhs_slave = self.discr_slave.assemble_matrix_rhs(
+            g_slave, data_slave
+        )
+        # The contact forces. Can be applied directly, these are in their own
+        # local coordinate systems.
+        cc[slave_ind, slave_ind] = traction_discr
+
+        # The contact condition discretization gives coefficients for the mortar
+        # variables. To finalize the relation with the contact conditions, we
+        # (from the right) 1) assign +- signs to the two sides of the mortar, so that
+        # summation in reality is a difference, 2) project to the mortar grid
+        # 3) project to the local coordinates of the fracture, 4) assign the
+        # coefficients of the displacement jump.
+        cc[slave_ind, mortar_ind] = (
+            displacement_jump_discr
+            * projection.project_tangential_normal(g_slave.num_cells)
+            * mg.mortar_to_slave_avg(nd=ambient_dimension)
+            * mg.sign_of_mortar_sides(nd=ambient_dimension)
+        )
+
+        # Right hand side system. In the local (surface) coordinate system.
+        rhs[slave_ind] = rhs_slave
+
+        ### Equation for the mortar rows
+
+        # This is first a stress balance: stress from the higher dimensional
+        # domain (both interior and bound_stress) should match with the contact stress:
+        # -\lambda_slave + \lambda_mortar = 0.
```

Comment:
I don't like the naming here. lambda should be reserved for the mortar variable (in this case displacement).
Also it should be a balance of the traction on the master and slave side, not the slave and the mortar:
- traction_slave + traction_master = 0

Reply:
The point is that we set actually set
T_mortar = T_slave +     and
T_mortar = - T_slave -. 
But I guess this should at least be made much clearer.

Reply:
Yes, the T_slave is compared to T_master through the mortar mappings, but there is no mortar representing traction. Or  do I misunderstand the formulation?

Reply:
1) The above should have been 
T_mortar + = T_slave and
T_mortar - = - T_slave,
where T_slave is a primary variable, and T_mortar reconstructed secondary variable (on the + and - side).

Reply:
You misunderstand my point.  T_mortar is not related to anything living on the mortar grid, thus, the naming T_mortar is deceiving. T_mortar is in fact the traction on the master projected on the mortar grid and T_master would therefore be a better name.

Reply:
Yes, you're right, I didn't quite get your point. I agree, and will change it accordingly.